packages/affiliates/libs/models.js

```diff
@@ -186,10 +186,21 @@ function History() {
     );
     return result;
   }
+  // get exact block number
+  function get(blockNumber) {
+    const result = history.find(x => x.blockNumber == blockNumber);
+    assert(result, "Not found in history: " + blockNumber);
+    return result;
+  }
+  function has(blockNumber) {
+    !!history.find(x => x.blockNumber == blockNumber);
```

Comment:
Why the double !?

Reply:
"has" technically should return a boolean, so that just converts truthy or falsey into a true or false

Reply:
there should be a return there though actually

Reply:
Side note: `!!(expression)` does the same thing as `Boolean(expression)`.